js/src/components/free-listings/configure-product-listings/checkErrors.test.js

```diff
@@ -0,0 +1,352 @@
+		let flatShipping;
+		let manualShipping;
+
+		beforeEach( () => {
+			flatShipping = { shipping_rate: 'flat' };
+			manualShipping = { shipping_rate: 'manual' };
+		} );
+
+		it( 'When the type of shipping rate is an invalid value, should not pass', () => {
+			// Not set yet
+			let errors = checkErrors( {}, [], [], [] );
+
+			expect( errors ).toHaveProperty( 'shipping_rate' );
+			expect( errors.shipping_rate ).toMatchSnapshot();
+
+			// Invalid value
+			errors = checkErrors( { shipping_rate: true }, [], [], [] );
+
+			expect( errors ).toHaveProperty( 'shipping_rate' );
+			expect( errors.shipping_rate ).toMatchSnapshot();
+
+			// Invalid value
+			errors = checkErrors( { shipping_rate: 'automatic' }, [], [], [] );
+
+			expect( errors ).toHaveProperty( 'shipping_rate' );
+			expect( errors.shipping_rate ).toMatchSnapshot();
+		} );
+
+		it( 'When the type of shipping rate is a valid value, should pass', () => {
+			// Selected flat
+			let errors = checkErrors( flatShipping, [], [], [] );
+
+			expect( errors ).not.toHaveProperty( 'shipping_rate' );
+
+			// Selected manual
+			errors = checkErrors( manualShipping, [], [], [] );
+
+			expect( errors ).not.toHaveProperty( 'shipping_rate' );
+		} );
+
+		describe( 'For flat type', () => {
+			function createFreeShipping( threshold ) {
+				return {
+					...flatShipping,
+					offers_free_shipping: true,
+					free_shipping_threshold: threshold,
+				};
+			}
+
+			it( `When there are any selected countries' shipping rates is not set, should not pass`, () => {
+				const rates = toRates( [ 'US', 10.5 ], [ 'FR', 12.8 ] );
+				const codes = [ 'US', 'JP', 'FR' ];
+
+				const errors = checkErrors( flatShipping, rates, [], codes );
+
+				expect( errors ).toHaveProperty( 'shipping_rate' );
+				expect( errors.shipping_rate ).toMatchSnapshot();
+			} );
+
+			it( `When all selected countries' shipping rates are set, should pass`, () => {
+				const rates = toRates( [ 'US', 10.5 ], [ 'FR', 12.8 ] );
+				const codes = [ 'US', 'FR' ];
+
+				const errors = checkErrors( flatShipping, rates, [], codes );
+
+				expect( errors ).not.toHaveProperty( 'shipping_rate' );
+			} );
+
+			it( `When there are any shipping rates is < 0, should not pass`, () => {
+				const rates = toRates( [ 'US', 10.5 ], [ 'JP', -0.01 ] );
+				const codes = [ 'US', 'JP' ];
+
+				const errors = checkErrors( flatShipping, rates, [], codes );
+
+				expect( errors ).toHaveProperty( 'shipping_rate' );
+				expect( errors.shipping_rate ).toMatchSnapshot();
+			} );
+
+			it( `When all shipping rates are ≥ 0, should pass`, () => {
+				const rates = toRates( [ 'US', 1 ], [ 'JP', 0 ] );
+				const codes = [ 'US', 'JP' ];
+
+				const errors = checkErrors( flatShipping, rates, [], codes );
+
+				expect( errors ).not.toHaveProperty( 'shipping_rate' );
+			} );
+
+			it( 'When the free shipping threshold is an invalid value, should not pass', () => {
+				// Not set yet
+				let freeShipping = createFreeShipping( null );
```

Comment:
:nail_care: 
For not set we could add also `undefined`

Reply:
Added a comment about the `null` value by https://github.com/woocommerce/google-listings-and-ads/pull/871/commits/a66e87b7f0db763b5de329109e31f0b93f26fb05